dotnet/src/SemanticKernel.Core/Memory/Whiteboard/WhiteboardBehavior.cs

```diff
@@ -0,0 +1,315 @@
+    /// </summary>
+    /// <returns>A task that completes when all messages are processed.</returns>
+    public async Task WhenProcessingCompleteAsync()
+    {
+        if (!this._updateWhiteboardTask.IsCompleted)
+        {
+            await this._updateWhiteboardTask.ConfigureAwait(false);
+        }
+    }
+
+    private async Task UpdateWhiteboardAsync(ChatMessage newMessage, CancellationToken cancellationToken = default)
+    {
+        var recentMessagesList = this._recentMessages.ToList();
+
+        // If there are more than MaxQueueSize messages in the queue, remove the oldest ones
+        for (int i = MaxQueueSize; i < recentMessagesList.Count; i++)
+        {
+            this._recentMessages.TryDequeue(out _);
+        }
+
+        // Extract the most important information from the input messages.
+        var basicMessages = recentMessagesList
+            .Select(m => new BasicMessage
+            {
+                AuthorName = m.AuthorName,
+                Role = m.Role.ToString(),
+                Text = m.Text
+            });
+
+        // Serialize the input messages and the current whiteboard content to JSON.
+        var inputMessagesJson = JsonSerializer.Serialize(basicMessages, WhiteboardBehaviorSourceGenerationContext.Default.IEnumerableBasicMessage);
+        var currentWhiteboardJson = JsonSerializer.Serialize(this._currentWhiteboardContent, WhiteboardBehaviorSourceGenerationContext.Default.ListString);
+
+        // Inovke the LLM to extract the latest information from the input messages and update the whiteboard.
+        var result = await this._chatClient.GetResponseAsync(
+            FormatPromptTemplate(inputMessagesJson, currentWhiteboardJson, this._maxWhiteboardMessages),
+            new()
+            {
+                Temperature = 0,
+                ResponseFormat = new ChatResponseFormatJson(s_structuredOutputSchema.RootElement),
+            },
+            cancellationToken).ConfigureAwait(false);
+
+        // Update the current whiteboard content with the LLM result.
+        var newWhiteboardResponse = JsonSerializer.Deserialize(result.ToString(), WhiteboardBehaviorSourceGenerationContext.Default.NewWhiteboardResponse);
+        this._currentWhiteboardContent = newWhiteboardResponse?.NewWhiteboard ?? [];
+    }
+
+    private static string FormatPromptTemplate(string inputMessagesJson, string currentWhiteboardJson, int maxWhiteboardMessages)
+    {
+        var sb = new StringBuilder(MaintenancePromptTemplate);
+        sb.Replace("{{$inputMessages}}", inputMessagesJson);
+        sb.Replace("{{$currentWhiteboard}}", currentWhiteboardJson);
+        sb.Replace("{{$maxWhiteboardMessages}}", maxWhiteboardMessages.ToString());
+        return sb.ToString();
+    }
+
+    /// <summary>
+    /// Gets the prompt template to use for maintaining the whiteboard.
+    /// </summary>
+    private const string MaintenancePromptTemplate =
```

Comment:
As a component I would allow the caller to improve/adjusting the prompt, suggest moving this prompt out of the code-base, keeping in a configuration or something similar.

```suggestion
    private const string MaintenancePromptTemplate =
```

Reply:
Sure, make it configurable by allowing it to be passed via options.